chapters/02-ruby_on_rails/readme.md

```diff
@@ -0,0 +1,795 @@
+much easier and more fun. It includes everything you need to build fantastic
+applications...* - [rubyonrails.org](https://rubyonrails.org)
+
+Rails is a web framework written for Ruby. This means that when you create a
+Rails project like we will today, you will have at your disposal all tools to
+display web pages, route users from one page to another, show data you have
+stored in database and many many more. 
+
+### Good documentation
+
+To give you a better perspective what are the basic principles on which Rails
+was designed, take a look on article written by one of its creators -
+[The Rails Doctrine by David Heinemeier Hansson](https://rubyonrails.org/doctrine/).
+And when you find yourself in a need of a guide or documentation, you can check
+out [official guides](https://guides.rubyonrails.org/) which unlike many others
+don't focus on explaining all the components like classes and methods but focus
+on solving problems you will encounter on day-to-day basis. And if you are then
+still looking for more insight in how things work there is an
+[API documentation](https://api.rubyonrails.org/).
+
+### The main principles
+
+With that out of the way let's take a look at two major guidlines for Rails
+development:
+
+* **Don't repeat yourself**, there is a specific place for everything. By not
+  writing the same information over and over again, our code is more
+  maintainable, more extensible, and less buggy.
+* Convention over configuration or bluntly said,
+  **"You're not a beautiful and unique snowflake"**. In most frameworks there is
+  a million of different ways how to do even basic tasks. If you Google a Rails
+  problem you probably won't be greeted with 20 completely different solutions
+  to your problem on Stack Overflow. You won't have to check which of these
+  solutions works best with the rest of your application. There actually is a
+  most convenient way how to do a certain thing. This convention means that you
+  can make stuff work the way you want very quickly and without knowing how it
+  works under the hood.
+
+![Rails doctrine](./images/doctrine.png)
+
+Rails is a web framework which strongly follows model-view-controller principle.
+This principle, as the name suggests, consists of three parts. **Models** are
+descriptions of your data objects (eg. post has id, author, date and content).
+**Controllers** implement business logic which controls your application,
+accesses data, and forwards stuff to the views and finally **views** are in
+charge of showing your page or API result to the users.
+
+## Ruby and gems
+
+If you have some experience with Python, PHP or Node, you might know of pip,
+composer or npm. All of these tools are package managers for their respective
+programming languages. They help you get all the tools that you need without
+copying source code to your project. Package manager for Ruby is called **gem**.
+
+```bash
```

Comment:
If they installed via rbenv, they already have the gem command.

Reply:
Will add now with an additional assignment.